lib/Listerine/channels.ex

```diff
@@ -157,12 +159,57 @@ defmodule Listerine.Channels do
     prepend = fn x, l -> [x | l] end
 
     course["channels"]
-    |> Enum.reduce([],
-      fn c, ac -> [Channel.get(c) |> do_or_nil.(&Channel.delete/1) | ac] end)
+    |> Enum.reduce(
+      [],
+      fn c, ac -> [Channel.get(c) |> do_or_nil.(&Channel.delete/1) | ac] end
+    )
     |> (fn
           nil -> nil
           a -> Enum.find(a, fn x -> x.type == 4 end).name
         end).()
     |> prepend.(remove_course_channels(others))
   end
+
+  def manage_roles(_message, [], _mode) do
+    []
+  end
+
+  def manage_roles(message, [name | tail], mode) do
+    guild = message.channel.guild_id
+    member = Guild.get_member(guild, message.author.id)
+    roles = get_courses()
+
+    roles =
+      Map.merge(roles["1"], roles["2"])
+      |> Map.merge(roles["3"])
+
+    modified =
```

Comment:
We could and should abstract this behaviour. 
A function should do one thing only, single responsibility principle.